pkg/sdk/tags_test.go

```diff
@@ -322,3 +297,97 @@ func TestTagAlter(t *testing.T) {
 		assertOptsInvalidJoinedErrors(t, opts, errExactlyOneOf("TagUnset", "MaskingPolicies", "AllowedValues", "Comment"))
 	})
 }
+
+func TestTagSet(t *testing.T) {
+	id := RandomSchemaObjectIdentifier()
+	defaultOpts := func() *setTagOptions {
+		return &setTagOptions{
+			objectType: ObjectTypeStage,
+			objectName: id,
+		}
+	}
+
+	t.Run("validation: nil options", func(t *testing.T) {
+		opts := (*setTagOptions)(nil)
+		assertOptsInvalidJoinedErrors(t, opts, ErrNilOptions)
+	})
+
+	t.Run("validation: incorrect identifier", func(t *testing.T) {
+		opts := defaultOpts()
+		opts.objectName = NewSchemaObjectIdentifier("", "", "")
+		assertOptsInvalidJoinedErrors(t, opts, ErrInvalidObjectIdentifier)
+	})
+
+	t.Run("set with all optional", func(t *testing.T) {
+		opts := defaultOpts()
+		opts.SetTags = []TagAssociation{
+			{
+				Name:  NewAccountObjectIdentifier("tag1"),
+				Value: "value1",
+			},
+		}
+		assertOptsValidAndSQLEquals(t, opts, `ALTER %s %s SET TAG "tag1" = 'value1'`, opts.objectType, id.FullyQualifiedName())
+	})
+
+	t.Run("set with column", func(t *testing.T) {
+		table, column := NewSchemaObjectIdentifier("db1", "schema1", "table1"), "column1"
+		objectName := NewObjectIdentifierFromFullyQualifiedName(fmt.Sprintf("%s.%s.%s.%s", table.DatabaseName(), table.SchemaName(), table.Name(), column))
```

Comment:
its really very useful in this case, as we need to do:

`ALTER TABLE "db1"."schema1"."table1" MODIFY COLUMN "column1" SET TAG "tag1" = 'value1'`

which as you can see, does not follow the table column identifier fully qualified name. I added it anyways, because its not a lot of work